### ISSUE TITLE
False positive protected-access on call from nested function

### ISSUE THREAD

**Issue by NeilGirdhar** [PyCQA/pylint#3546]
Should it be okay to call a protected function from within a nested function?
```python
class X:
    def _f(self):
        return 0

    def g(self):
        def internal(x: X):
            return x._f()
        return internal(self)
```
gives `protected-access`

**Comment by PCManticore** [comment]
Yes, it's fine to emit this error, even if it's an internal function. pylint is not capable of figuring out yet that `x` is of the same type and is the same instance like `self`.

**Comment by NeilGirdhar** [comment]
Does pylint not see the type annotation?
